diff --git a/vm_plugin.py b/vm_plugin.py
index 76c7f54..3b6d9d1 100644
--- a/vm_plugin.py
+++ b/vm_plugin.py
@@ -5577,7 +5577,7 @@ class Vm(object):
                 return disk
 
             def filebased_volume(_dev_letter, _v):
-                disk = etree.Element('disk', {'type': 'file', 'device': 'disk', 'snapshot': 'external'})
+                disk = etree.Element('disk', {'type': 'file', 'device': 'disk'})
                 driver_elements = {'name': 'qemu', 'type': linux.get_img_fmt(_v.installPath), 'cache': _v.cacheMode, 'discard': 'unmap'}
                 if cmd.addons and cmd.addons['useDataPlane'] is True:
                     driver_elements['dataplane'] = 'on'
@@ -5603,7 +5603,7 @@ class Vm(object):
                     e(disk, 'wwn', _v.wwn)
                     return disk
 
-                if _v.useVirtio:
-                    e(disk, 'target', None, {'dev': 'vd%s' % _dev_letter, 'bus': 'virtio'})
+                if _v.useVirtio:
+                    e(disk, 'target', None, {'dev': 'vd%s' % _dev_letter, 'bus': 'scsi'})
                 else:
                     dev_format = Vm._get_disk_target_dev_format(default_bus_type)
                     e(disk, 'target', None, {'dev': dev_format % _dev_letter, 'bus': default_bus_type})
